src/hostname/hostnamed.c

```diff
@@ -597,6 +604,59 @@ static int method_set_location(sd_bus_message *m, void *userdata, sd_bus_error *
         return set_machine_info(userdata, m, PROP_LOCATION, method_set_location, error);
 }
 
+static bool uuid_is_valid(sd_id128_t id) {
+        unsigned i;
+
+        /* Here we reject uuid if all bytes of the uuid are the same,
+         * e.g., 0000..., 3a3a3a..., or ffffff....*/
+
+        for (i = 1; i < 16; i++)
+                if (id.bytes[i] != id.bytes[0])
+                        return true;
+
```

Comment:
I don't like this arbitrary check. Let's not overthink this. This is just a helper to read a privileged file, and it shouldn't decide if the UUID is good or not. The user can do do that, depending on what it is to be used for. (For example, I can imagine the "about this machine" dialogue in gnome displaying the product UUID using this at some point, and then it should be displayed as is, whatever the manufacturer put there.) The reason for using the UUID is that it provides stability, and other DHCP implementations will use the UUID as is, and so should we. I guess it'd be OK to filter out all 0's on the client side, but that's about it.

Reply:
OK. The check is now moved to networkd.